### ISSUE TITLE
Add ability to log raw response body

### ISSUE THREAD

**Issue by Darkvater** [quarkusio/quarkus#33540]
### Description

With the use of `ServerResponseContextFilter` it is easy to add some custom logging of the response headers, body, duration (with a little bit of magic), etc.

However the response body available through `ctx.entity` is the actual object, not what has been written to the output. As such there is the "danger" of logging something else than what has been sent to the client.

### Implementation ideas

On the request level this is solved by using a callback on `HttpServerRequest.body().onComplete` so maybe something similar can be done for response?

In spring-boot the same can be achieved by using `org.sprimgframework.web.util.ContentCachingResponseWrapper.contentAsByteArray` or a `MultiOutputServletResponse` that wraps the output stream into a DelegatingServletOutputStream with TeeOutputStream.

**Comment by geoand** [comment]
Does `jakarta.ws.rs.ext.WriterInterceptor` fit your needs?

**Comment by geoand** [comment]
> More importantly the context of WriterInterceptorContext doesn't give you access to otherlogging essentials such as HTTP method, URI

You should be able to use `@Context` to inject this sort of information into the appropriate fields.

**Comment by Darkvater** [comment]
yes, that would be a potential approach to getting the other data, however it will not help with the complexity of doing a (custom) logging of the response entity.
